mito-ai/mito_ai/utils/gemini_utils.py

```diff
@@ -17,15 +18,18 @@
 
 def _prepare_gemini_request_data_and_headers(
         model: str,
-        contents: str,
+        contents: Sequence[Content],
         message_type: MessageType,
         response_format_info: Optional[Any] = None,
+        system_prompt: Optional[str] = None,
 ) -> Tuple[Dict[str, Any], Dict[str, str]]:
     inner_data = {
         "model": model,
-        "contents": contents,
+        "contents": [{"role": c.role, "parts": [{"text": p.text} for p in c.parts if p.text is not None]} for c in contents if c.parts],
```

Comment:
Why do we need to do this? The `convert_messages_for_gemini` function should be getting the messages into the right format so we don't have to do any more preprocessing of it, right?